bin/node/runtime/src/lib.rs

```diff
@@ -201,57 +201,165 @@ parameter_types! {
 
 const_assert!(NORMAL_DISPATCH_RATIO.deconstruct() >= AVERAGE_ON_INITIALIZE_RATIO.deconstruct());
 
-pub struct UnpausablePallets;
-impl Contains<pallet_tx_pause::PalletNameOf<Runtime>> for UnpausablePallets {
-	fn contains(pallet: &pallet_tx_pause::PalletNameOf<Runtime>) -> bool {
-		pallet.as_ref() ==
-			<pallet_safe_mode::Pallet<Runtime> as PalletInfoAccess>::name()
-				.as_bytes()
-				.to_vec()
+/// Filter to block balance pallet calls
+/// Used for both SafeMode and TxPause pallets
+/// Therefor we include both so they cannot affect each other
+pub struct UnfilterableCalls;
+impl Contains<RuntimeCall> for UnfilterableCalls {
+	fn contains(call: &RuntimeCall) -> bool {
+		match call {
+			RuntimeCall::System(_) | RuntimeCall::SafeMode(_) | RuntimeCall::TxPause(_) => true,
+			RuntimeCall::Balances(_) => false,
+			_ => false,
+		}
 	}
 }
 
 impl pallet_tx_pause::Config for Runtime {
-	type Event = Event;
-	type UnpausablePallets = UnpausablePallets;
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeCall = RuntimeCall;
 	type PauseOrigin = EnsureRoot<AccountId>;
 	type UnpauseOrigin = EnsureRoot<AccountId>;
+	type UnfilterableCalls = UnfilterableCalls;
 	type MaxNameLen = ConstU32<256>;
 	type PauseTooLongNames = ConstBool<true>;
 	type WeightInfo = pallet_tx_pause::weights::SubstrateWeight<Runtime>;
 }
 
-parameter_types! {
-	// signed config
-	pub const ActivateStakeAmount: Balance = 1 * DOLLARS; //TODO This needs to be something sensible for the implications of enablement!
-	pub const ExtendStakeAmount: Balance = 1 * DOLLARS; //TODO This needs to be something sensible for the implications of enablement!
-	pub BlockHeight: BlockNumber = System::block_number(); // TODO ensure this plus config below is correct
+/// An origin that can enable the safe-mode by force.
+pub enum ForceActivateOrigin {
+	Weak,
+	Medium,
+	Strong,
+}
+
+/// An origin that can extend the safe-mode by force.
+pub enum ForceExtendOrigin {
+	Weak,
+	Medium,
+	Strong,
+}
+
+impl ForceActivateOrigin {
+	/// The duration of how long the safe-mode will be activated.
+	pub fn duration(&self) -> u32 {
```

Comment:
I will prefer to make it 
```
/// An origin that can enable the safe-mode by force.
 pub enum ForceActivateOrigin {
 	Weak = 5,
 	Medium = 7,
 	Strong = 11,
 }
```

so that frontend doesn't need to hardcod those number once more

Reply:
So TIL https://doc.rust-lang.org/reference/items/enumerations.html#custom-discriminant-values-for-fieldless-enumerations

So [this works](https://github.com/paritytech/substrate/compare/ds/tmp-sm-txp...ds/tmp-enum-origin?expand=1#diff-c8a277fd45ad463eff816d0cd84b815aab8ddca058f250342624ced397d4c46dR92) to yield an integer castable item, so BlockNumber works fine.

But you cannot assign the _same_ value for any origin this way as it would conflict:
```rust
enum SharedDiscriminantError {
    SharedA = 1,
    SharedB = 1
}
```

I think this restriction would be OK, but hope there is a better method to get there... Open to ideas.